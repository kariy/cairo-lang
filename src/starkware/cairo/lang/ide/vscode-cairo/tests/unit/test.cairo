# SYNTAX TEST "source.cairo" "sample testcase"

func foo{output_ptr: felt*}(a: felt*):
# <---- storage.type.function.cairo
#   ^ meta.function.cairo
#    ^^^ entity.name.function.cairo
#        ^^^ variable.parameter.implicit.cairo
#                           ^^^^^^^^ meta.parameter.cairo 
#                                    ^ meta.function.definition.end.cairo
    if 0 == 0:
        let x = 10
    else:
    end

    tempvar x = 10
#   ^^^^^^^ storage.type.cairo
#               ^^ constant.numeric.decimal.cairo
end
# <--- keyword.control.other.cairo