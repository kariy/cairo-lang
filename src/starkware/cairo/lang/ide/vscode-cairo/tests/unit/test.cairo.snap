># SYNTAX TEST "source.cairo" "sample testcase"
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.cairo comment.line.number-sign.cairo
>
>func foo{output_ptr: felt*}(a: felt*):
#^^^^ source.cairo meta.function.cairo storage.type.function.cairo
#    ^ source.cairo meta.function.cairo
#     ^^^ source.cairo meta.function.cairo meta.function.definition.cairo entity.name.function.cairo
#        ^ source.cairo meta.function.cairo meta.function.definition.cairo meta.parameter.implicit.cairo
#         ^^^^^^^^^^ source.cairo meta.function.cairo meta.function.definition.cairo meta.parameter.implicit.cairo variable.parameter.implicit.cairo
#                   ^^ source.cairo meta.function.cairo meta.function.definition.cairo meta.parameter.implicit.cairo
#                     ^^^^ source.cairo meta.function.cairo meta.function.definition.cairo meta.parameter.implicit.cairo support.type.primitive.cairo
#                         ^ source.cairo meta.function.cairo meta.function.definition.cairo meta.parameter.implicit.cairo
#                          ^ source.cairo meta.function.cairo meta.function.definition.cairo meta.parameter.implicit.cairo
#                           ^ source.cairo meta.function.cairo meta.function.definition.cairo meta.parameter.cairo
#                            ^ source.cairo meta.function.cairo meta.function.definition.cairo meta.parameter.cairo variable.parameter.cairo
#                             ^^ source.cairo meta.function.cairo meta.function.definition.cairo meta.parameter.cairo
#                               ^^^^ source.cairo meta.function.cairo meta.function.definition.cairo meta.parameter.cairo support.type.primitive.cairo
#                                   ^ source.cairo meta.function.cairo meta.function.definition.cairo meta.parameter.cairo
#                                    ^ source.cairo meta.function.cairo meta.function.definition.cairo meta.parameter.cairo
#                                     ^ source.cairo meta.function.cairo meta.function.definition.cairo meta.function.definition.end.cairo
>
>    tempvar x = 10
#^^^^ source.cairo meta.function.cairo
#    ^^^^^^^ source.cairo meta.function.cairo meta.function.definition.cairo entity.name.function.cairo
#           ^^^^^^^^ source.cairo meta.function.cairo meta.function.definition.cairo
>
>end
#^^^^ source.cairo meta.function.cairo meta.function.definition.cairo
>
>